README.md

```diff
@@ -66,55 +66,65 @@ flowchart LR
     Handle -->|Decode Init Errors| Decoder
 
     Context -->|Run| Library
-    Context -->|Store Go References| ContextAllocator
+    Context -->|Store Go References| CGORefPool
 
-    Encoder -->|Allocate Waf Objects| EncoderAllocator
+    Encoder -->|Allocate Waf Objects| TempCGORefPool
 
-    EncoderAllocator -->|Copy after each encoding| ContextAllocator
+    TempCGORefPool -->|Copy after each encoding| CGORefPool
 
     Library -->|Call C code| libddwaf
 
     classDef hidden display: none;
 ```
 
-### Allocator
+### CGO Reference Pool
 
-The cgoRefPool is a pure Go cgoRefPool of `ddwaf_object` C values on the Go memory heap.
-the `cgoRefPool` go type is a way to make sure we can safely send go allocated data to the C side of the WAF
+The cgoRefPool type is a pure Go pointer pool of `ddwaf_object` C values on the Go memory heap.
+the `cgoRefPool` go type is a way to make sure we can safely send Go allocated data to the C side of the WAF
```

Comment:
Sometimes it's "go type", sometimes just type... I'd recommend making all "type" implicitly be go and to actually expressly specify when you're referring to a C type...

There is also a lack of punctuation at the end of this sentence...

Reply:
```suggestion
The cgoRefPool type is a pure Go pointer pool of `ddwaf_object` C values on the Go memory heap.
It is a way to make sure we can safely send Go allocated data to the C side of the WAF without the Garbage Collector reclaiming the memory too early.
```

Do not hesitate to write suggestions in case like this, it's easier for everyone :+1:

Reply:
I do when I have a clear & confident idea of what I'd like to see... in this case I felt like this was very opinion land and I was also not sure about the rest of the doc just yet so it could also have been a bad idea